addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,810 @@
+                                                    **request_data.get('header')
+                                                }, timeout=(30, 30))
+        except (ValueError, requests.exceptions.RequestException) as ex:
+            return {
+                'error': str(ex),
+                'blocking_level': 'warning',
+                'excepted': True
+            }
+
+        # Authentication errors do not return json
+        if request_response.status_code == 401:
+            return {
+                'error': _("API %s could not be authenticated") % request_url,
+                'blocking_level': 'error'
+            }
+
+        if request_response.status_code in (303, 400, 404, 500, 409, 502, 503):
+            return {'error': request_response.text, 'blocking_level': 'error'}
+
+        try:
+            response_data = request_response.json()
+        except json.decoder.JSONDecodeError as e:
+            return {
+                'error': _("JSON response from ZATCA could not be decoded"),
+                'blocking_level': 'error'
+            }
+
+        if not request_response.ok and (response_data.get('errors') or response_data.get('warnings')):
+            if isinstance(response_data, dict) and response_data.get('errors'):
+                return {
+                    'error': _("Invoice submission to ZATCA returned errors"),
+                    'json_errors': response_data['errors'],
+                    'blocking_level': 'error',
+                }
+            return {
+                'error': request_response.reason,
+                'blocking_level': 'error'
+            }
+        return response_data
+
+    def _l10n_sa_api_headers(self):
+        """
+            Return the base headers to be included in ZATCA API calls
+        :return:
+        """
+        return {
+            'Content-Type': 'application/json',
+            'Accept-Language': 'en',
+            'Accept-Version': 'V2'
+        }
+
+    def _l10n_sa_authorization_header(self, CSID_data):
+        """
+            Compute the Authorization header by combining the CSID and the Secret key, then encode to Base64
+        :param CSID_data: Either CCSID or PCSID data
+        :return: Authorization Header
+        """
+        auth_str = "%s:%s" % (CSID_data['binarySecurityToken'], CSID_data['secret'])
+        return 'Basic ' + b64encode(auth_str.encode()).decode()
+
+    def _l10n_sa_assert_clearance_status(self, invoice, clearance_data):
+        """
+            Assert Clearance status. To be overridden in case there are any other cases to be accounted for
+        :param invoice: Cleared invoice
+        :param clearance_data: Results of the clearance process
+        """
+        if clearance_data.get('clearanceStatus', '') != 'CLEARED':
+            return {'error': _("Invoice could not be cleared: \r\n %s ") % clearance_data, 'blocking_level': 'error'}
+        return clearance_data
+
+    # ====== API Calls to ZATCA =======
```

Comment:
This API Calls is like everything mixed together?  Why not put the ones for compliance, ...  in the account_journal and restrict to the XML generation here?